### DISCUSSION TITLE
Methods `orWhereKey()` and `oWhereKeyNot()` in Eloquent Builder

### DISCUSSION THREAD

**Discussion by GuNZfREE** [laravel/framework#52777] [Ideas]
Hello!

The [`whereKey()`](https://github.com/laravel/framework/blob/54cf9db8a57744f4c6f755a0bedc222a4d9cd664/src/Illuminate/Database/Eloquent/Builder.php#L245) and [`whereKeyNot()`](https://github.com/laravel/framework/blob/54cf9db8a57744f4c6f755a0bedc222a4d9cd664/src/Illuminate/Database/Eloquent/Builder.php#L274) methods from `Eloquent\Builder` were added in Laravel 5.x, and I love using them because of the ability to pass different data types (models, arrays, etc) .
But for all this time I could not find a PR that would suggest adding the `orWhereKey()` and `orWhereKeyNot()` methods for "or where" clause. 
Or am I the only one missing these methods? :D

I think this would be a useful new syntactic sugar for many.

```php
# Before
$user = User::query()
	->orWhere(
		fn(Builder $builder) => $builder->whereKey($arrayIds)
	)
	->orWhereHas('foo', function (Builder $builder) {
		$builder
			->where...()
			->....
	});

# After
$user = User::query()
	->orWhereKey($arrayIds)
	->orWhereHas('foo', function (Builder $builder) {
		$builder
			->where...()
			->....
	});
```

Thanks!

**Comment by ghost** [comment]
Hi. orWhere can lead to data leaks if not used properly. You can create macros and use those.
